tests/common/test_persist_firewall_rules.py

```diff
@@ -0,0 +1,297 @@
+        self.assertTrue(os.path.exists(self._network_service_unit_file), "Service unit file should be there")
+        self.assertTrue(os.path.exists(self._drop_in_file), "Drop in file should be there")
+        self.assertTrue(os.path.exists(self._network_service_bin_file), "Network setup Binary file should be there")
+
+    @staticmethod
+    def __mock_network_setup_service_enabled(cmd):
+        if "firewall-cmd" in cmd:
+            return True, ["echo", "not-running"]
+        if "systemctl" in cmd:
+            return True, ["echo", "enabled"]
+        return False, []
+
+    @staticmethod
+    def __mock_network_setup_service_disabled(cmd):
+        if "firewall-cmd" in cmd:
+            return True, ["echo", "not-running"]
+        if "systemctl" in cmd:
+            return True, ["echo", "not enabled"]
+        return False, []
+
+    def test_it_should_skip_setup_if_firewalld_already_enabled(self):
+        self.__replace_popen_cmd = lambda cmd: ("firewall-cmd" in cmd, ["echo", "running"])
+        with self._get_persist_firewall_rules_handler() as handler:
+            handler.setup()
+
+        # Assert we verified that rules were set using firewall-cmd
+        self.__assert_firewall_called(cmd=FirewallCmdDirectCommands.QueryPassThrough, validate_command_called=True)
+        # Assert no commands for adding rules using firewall-cmd were called
+        self.__assert_firewall_called(cmd=FirewallCmdDirectCommands.PassThrough, validate_command_called=False)
+        # Assert no commands for systemctl were called
+        self.assertFalse(any(["systemctl" in cmd for cmd in self.__executed_commands]), "Systemctl shouldn't be called")
+
+    def test_it_should_skip_setup_if_agent_network_setup_service_already_enabled(self):
+        self.__replace_popen_cmd = TestPersistFirewallRulesHandler.__mock_network_setup_service_enabled
+        with self._get_persist_firewall_rules_handler() as handler:
+            handler.setup()
+
+        self.__assert_systemctl_called(cmd="is-enabled", validate_command_called=True)
+        self.__assert_systemctl_called(cmd="enabled", validate_command_called=False)
+        self.__assert_firewall_cmd_running_called(validate_command_called=True)
+        self.__assert_firewall_called(cmd=FirewallCmdDirectCommands.QueryPassThrough, validate_command_called=False)
+        self.__assert_firewall_called(cmd=FirewallCmdDirectCommands.PassThrough, validate_command_called=False)
+
+    def test_it_should_always_replace_only_drop_in_file_if_using_custom_network_service(self):
+        test_str = 'Environment="DST_IP={0}" "UID={1}" "WAIT={2}"'
+        self.__replace_popen_cmd = TestPersistFirewallRulesHandler.__mock_network_setup_service_disabled
+        with self._get_persist_firewall_rules_handler() as handler:
+            self.assertFalse(os.path.exists(self._drop_in_file), "Drop in file should not be there")
+            self.assertFalse(os.path.exists(self._network_service_unit_file), "Unit file not present")
```

Comment:
The message ("Unit file not present") seems to imply that the file should be present, but we are asserting it is not

Reply:
Ohh yes good catch! Let me fix the message.